packages/engine/Source/Scene/Model/ModelDrawCommands.js

```diff
@@ -21,17 +31,80 @@ import ModelDrawCommand from "./ModelDrawCommand.js";
  *
  * @param {PrimitiveRenderResources} primitiveRenderResources The render resources for a primitive.
  * @param {FrameState} frameState The frame state for creating GPU resources.
- *
  * @returns {ModelDrawCommand|ClassificationModelDrawCommand} The generated ModelDrawCommand or ClassificationModelDrawCommand.
  *
  * @private
  */
-function buildDrawCommand(primitiveRenderResources, frameState) {
+ModelDrawCommands.buildModelDrawCommand = function (
+  primitiveRenderResources,
+  frameState
+) {
   const shaderBuilder = primitiveRenderResources.shaderBuilder;
+  const shaderProgram = ModelDrawCommands.createShaderProgram(
+    primitiveRenderResources,
+    shaderBuilder,
+    frameState
+  );
+
+  const command = ModelDrawCommands.buildDrawCommandForModel(
+    primitiveRenderResources,
+    shaderProgram,
+    frameState
+  );
+
+  const model = primitiveRenderResources.model;
+  const hasClassification = defined(model.classificationType);
+  if (hasClassification) {
+    return new ClassificationModelDrawCommand({
+      primitiveRenderResources: primitiveRenderResources,
+      command: command,
+    });
+  }
+
+  return new ModelDrawCommand({
+    primitiveRenderResources: primitiveRenderResources,
+    command: command,
+  });
+};
+
+/**
+ * @private
+ */
+ModelDrawCommands.createShaderProgram = function (
```

Comment:
As far as I can see, only the `buildModelDrawCommand` method is used outside this file. Is there a reason to expose the local helper functions?

Reply:
This may be the result of two things:

- I don't like free-floating functions that belong nowhere
  - Why? For many reasons. For example: **Iff** one of these functions should be explicitly unit-tested, it will have to be exposed
    - How? A full-text search for `"for testing"` gives the answer...
- There is no proper support for scoping and visibility in JS

However, as long as there is no clear guideline about this (i.e. no "best practices" in terms of API- and "class"-structure  design in the Coding Guide), I have to consider this as subjective, and now changed them to be local functions...